### ISSUE TITLE
External server vs subdomains

### ISSUE THREAD

**Issue by baldurpan** [bdkjones/CodeKit2#482]
Been having trouble with CodeKit on the current project I'm working on.

CodeKit server is set up on baldur.local:5757 and has the external server pointed to baldur.dev.myproject.is.

As we know, CodeKit replaces all the links automatically to keep things smoothly running between pages. This is all good.
However, I also have set upp a subdomain for static files on static.baldur.dev.myproject.is and Codekit seems to change those links to F.e. static.baldur.local:5757/myfile.js which, of course, becomes a dead link.

**Comment by bdkjones** [comment]
Yep. CodeKit simply replaces _every_ instance of the external server name that it finds on the page, using simple string-replacement. It does not parse the HTML (because that would be very slow). CodeKit's proxy server isn't meant for production, so there are edge cases like this that a _real_ production proxy server would handle but that CodeKit will miss. The tradeoff is speed: CodeKit's proxy is much faster.

**Comment by baldurpan** [comment]
So it is not possible to only replace instances that don't have a dot in front of them?
I wouldn't think that a simple regex replace would hurt the performance that much.

Having it as an option would be nice at least.

**Comment by bdkjones** [comment]
Ha. Regex is not available in C, which is what CodeKit's server is written in. Regex is also very bad for performance --- there's a reason C is hundreds of times faster than JavaScript. However, I agree that the subdomain problem is worth working around, so I will implement an additional check for it.

**Comment by baldurpan** [comment]
Pardon my ignorance, but my coding skills are limited to web developing. :)
Appreciate the checkup. I'm sure more people will hit this wall sooner or later.
